applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_startegy.h

```diff
@@ -0,0 +1,576 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla, Eduard Gómez
+//
+//
+
+#if !defined(KRATOS_COMPRESSIBLE_NAVIER_STOKES_EXPLICIT_SOLVING_STRATEGY_BASE)
+#define KRATOS_COMPRESSIBLE_NAVIER_STOKES_EXPLICIT_SOLVING_STRATEGY_BASE
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/global_variables.h"
+#include "includes/model_part.h"
+#include "utilities/atomic_utilities.h"
+#include "utilities/math_utils.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+#include "custom_processes/shock_capturing_physics_based_process.h"
+#include "custom_processes/shock_capturing_entropy_viscosity_process.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/** @brief Explicit solving strategy base class
+ * @details This is the base class from which we will derive all the explicit strategies
+ * for the compressible Navier Stokes equations (BFECC, RK4, ...)
+ */
+template <typename TBaseExplicitStartegy>
+class CompressibleNavierStokesExplicitSolvingStrategy
+: public TBaseExplicitStartegy
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// The base class definition
+
+    typedef TBaseExplicitStartegy BaseType;
+
+    /// The explicit builder and solver definition
+    typedef typename BaseType::ExplicitBuilderType ExplicitBuilderType;
+
+    /// The local vector definition
+    typedef typename TBaseExplicitStartegy::LocalSystemVectorType LocalSystemVectorType;
+
+    /// Pointer definition of CompressibleNavierStokesExplicitSolvingStrategy
+    KRATOS_CLASS_POINTER_DEFINITION(CompressibleNavierStokesExplicitSolvingStrategy);
+
+    /// Local Flags
+    KRATOS_DEFINE_LOCAL_FLAG(SHOCK_CAPTURING);
```

Comment:
Why is this flag required? As far as I see you're checking the shock capturing with the nullity of the shock capturing pointer.

Reply:
This was lifted from the old implementation of the Runge-Kutta for compressible Navier Stokes. Now that you mention it, it isn't used anywhere so it can be removed.